### PR TITLE
fix menu mapping

### PR DESCRIPTION
**Pull request by klas**
Fix all new ids beeing 0 in #__jupgradepro_menus, causing mutiple problem including lost module assignments

### CODE REVIEW

**Comment by klas** [comment]
haven't checked other versions schemas...

**Comment by klas** [comment]
I updated the pull -  menus nesting was broken too as 
a) you can't get parent id before all ids are inserted, child can have lower id than parent e.g. when menu item was moved to another parent
b) search by old alias does not work reliably as aliases are often changed to avoid duplicates

He above was causing a lot of menus beeing moved to root.

I also added table rebuild() function to menus and categories to ensure whole tree is consistent.

File: libraries/matware/jupgrade/schemas/1.5/categories.php [hunk 1]
```diff
@@ -154,7 +154,7 @@ public function dataHook($rows = null)
 			}
 
 			// Reset some fields
-			$category['asset_id'] = $category['lft'] = $category['rgt'] = null;
+			$category['asset_id'] = $category['lft'] = $category['rgt'] =  null;
```
Comment (fastslack) [inline libraries/matware/jupgrade/schemas/1.5/categories.php]: @klas any reason to change this line?

**Comment by fastslack** [comment]
@klas dont worry about other versions because your change only affect Joomla! 1.5 source to 2.5+ destination

File: libraries/matware/jupgrade/schemas/1.5/categories.php (hunk 1 above)
Comment (klas) [inline libraries/matware/jupgrade/schemas/1.5/categories.php]: Not really :) Comment block must have been changed by phpStorm, same for other spaces changes in this pull..

**Comment by klas** [comment]
What I was trying to say is that other versions schemas perhaps need same change

File: libraries/matware/jupgrade/schemas/1.5/categories.php [hunk 2]
```diff
@@ -188,6 +188,16 @@ public function dataHook($rows = null)
 			$this->_step->_nextID($total);
 		}
 
+		// rebuild table
```
Comment (fastslack) [inline libraries/matware/jupgrade/schemas/1.5/categories.php]: JTable->rebuild() is already running at the end of each step https://github.com/fastslack/matware-libraries/blob/master/libraries/matware/jupgrade/category.php#L127

Comment (klas) [inline libraries/matware/jupgrade/schemas/1.5/categories.php]: Ok, I'll remove this.

Comment (klas) [inline libraries/matware/jupgrade/schemas/1.5/categories.php]: done

**Comment by fastslack** [comment]
Thanks @klas. I ll check others schemas more detailed

**Pull request merged**